src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+
+type Parent = 'father' | 'mother';
+
+const getParent = (
+  people: Person[],
+  person: Person,
+  parent: Parent,
+) => {
+  if (parent === 'father') {
+    return people.find(({ name }) => name === person.fatherName);
+  }
+
+  return people.find(({ name }) => name === person.motherName);
+};
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isPending, setIsPending] = useState(true);
+
+  const preparePeople = () => {
+    setIsPending(true);
+
+    getPeople().then(result => {
+      const prepared = result.map(person => ({
+        ...person,
+        father: getParent(result, person, 'father'),
+        mother: getParent(result, person, 'mother'),
```

Comment:
as an option, you can get rid of Parent type and getParent function:
```suggestion
        father: people.find(({ name }) => name === person.fatherName),
        mother: people.find(({ name }) => name === person.motherName),
```